### PR TITLE
Add enterprise edition and licensing

### PR DESCRIPTION
**Pull request by timgl**
## Changes

- Add an "ee" folder that is copyrighted (as opposed to our MIT licensed repository)
- You can remove this folder and the system will continue functioning normally, in case you want a purely MIT licensed product.
- Added a section in settings where you can add and manage license keys
- Created a repo and service to create licenses ([Internal link to repo](http://github.com/posthog/license) to see how to create a new license key).
- Add methods to the User model to expose which features are available (given the plan that is activated)

image

## Checklist

- [x] All querysets/queries filter by Team (if this PR affects any querysets/queries)
- [x] Backend tests (if this PR affects the backend)
- [ ] Cypress E2E tests (if this PR affects the front and/or backend)

### CODE REVIEW

File: ee/LICENSE [hunk 1]
```diff
@@ -0,0 +1,36 @@
+The PostHog Enterprise license (the “Enterprise License”)
+Copyright (c) 2020 PostHog Inc.
```
Comment (timgl) [inline ee/LICENSE:2]: @jamesefhawkins do you mind checking the changes to the licensing

File: LICENSE [hunk 2]
```diff
@@ -1,6 +1,10 @@
-The MIT License (MIT)
+Copyright (c) 2020 PostHog Inc.
 
-Copyright (c) 2020 PostHog (part of Hiberly Inc)
+Portions of this software are licensed as follows:
+
+* All content that resides under the "ee/" directory of this repository, if that directory exists, is licensed under the license defined in "growth/LICENSE".
```
Comment (mariusandra) [inline LICENSE:5]: `growth/LICENSE` --> `ee/LICENSE` ?

File: frontend/src/scenes/setup/licenseLogic.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    actions: {
+        setError: (error: Error) => ({ error }),
+        addLicense: (license: License) => ({ license }),
+        createLicense: (license: License) => ({ license }),
+    },
+    loaders: {
+        licenses: [
+            [],
+            {
+                loadEvents: async () => {
```
Comment (mariusandra) [inline frontend/src/scenes/setup/licenseLogic.ts:24]: Would this look better as `loadLicenses`?

**Review by mariusandra — APPROVED** [review]
I found two "typos" that could be fixed... and it might be good to run `kea-typegen` to get types for the new logic. The rest looks good!

Running it with and without the `"ee"` folder worked. However when running without that folder, the `/setup/licenses` page would still open when I entered the URL directly. Then it would crash when I tried to use it.

This can absolutely be left for a future PR, but I think we need a way to also split the frontend into the "ee" and "non-ee" parts, possibly moving the /setup/licenses page into a `ee/frontend` folder.

**Comment by timgl** [comment]
@mariusandra Yep agree 100%. I don't quite have the webpack wizardry (and we only have a tiny bit of ee frontend code right now). Added #1395, would love your input

**Comment by Twixes** [comment]
What's the use case for multiple licenses per team?

**Comment by timgl** [comment]
@Twixes You might add a license that expires, or you might add a license that has a higher plan than the one you currently have. It'll be unlikely you'll have multiple licenses active at once, but at least this gives you the history

**Pull request merged**